.github/workflows/ci-build.yaml

```diff
@@ -17,8 +17,97 @@ permissions:
   contents: read
 
 jobs:
+  changed-files:
+    name: Get changed files
+    outputs:
+      # reference: https://github.com/tj-actions/changed-files#outputs-
+      tests: ${{ steps.changed-files.outputs.tests_any_modified == 'true' }}
+      e2e-tests: ${{ steps.changed-files.outputs.e2e-tests_any_modified == 'true' }}
+      codegen: ${{ steps.changed-files.outputs.codegen_any_modified == 'true' }}
+      lint: ${{ steps.changed-files.outputs.lint_any_modified == 'true' }}
+      ui: ${{ steps.changed-files.outputs.ui_any_modified == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 50 # assume PRs are less than 50 commits
+      - name: Get relevant files changed per group
+        id: changed-files
+        uses: tj-actions/changed-files@v39
+        with:
+          files_yaml: |
+            common: &common
+              - .github/workflows/ci-build.yaml
```

Comment:
we may want to add `Makefile` and `tasks.yaml` here, but I wanted to avoid excessive testing. It may be better to be more conservative in case of a breaking `Makefile` change, for example, but those are rarely done in isolation

Reply:
In my experience, the Makefile doesn't change that frequently and I think it is better to err on the side of safety and have this run test. 

I was actually quite reliant on the testing when I did my nix stuff.

Reply:
> I was actually quite reliant on the testing when I did my nix stuff.

Not really impactful on the decision, but CI doesn't use Nix? So I am a bit confused by this statement, how would CI be helpful for Nix changes?

Reply:
I had to make changes to the Makefile and was reliant on the CI to check if it retained its previous functionality

Reply:
Oh for regression testing, gotcha

Reply:
Added both `Makefile` and `tasks.yaml` for the safest option